sw/host/spiflash/verilator_spi_interface.cc

```diff
@@ -91,21 +92,34 @@ bool VerilatorSpiInterface::Init() {
   return true;
 }
 
-bool VerilatorSpiInterface::TransmitFrame(const uint8_t *tx, uint8_t *rx,
-                                          size_t size) {
+bool VerilatorSpiInterface::TransmitFrame(const uint8_t *tx, size_t size) {
   size_t bytes_written = write(fd_, tx, size);
   if (bytes_written != size) {
     std::cerr << "Failed to write bytes to spi interface. Bytes written: "
               << bytes_written << " expected: " << size << std::endl;
     return false;
   }
   usleep(kWriteReadDelay);
-  size_t bytes_read = ReadBytes(fd_, rx, size);
+  return true;
+}
+
+bool VerilatorSpiInterface::CheckHash(const uint8_t *tx, size_t size) {
+  uint8_t hash[SHA256_DIGEST_LENGTH];
+  SHA256_CTX sha256;
+  SHA256_Init(&sha256);
+  SHA256_Update(&sha256, tx, size);
+  SHA256_Final(hash, &sha256);
+
+  uint8_t *rx = (uint8_t *)malloc(size);
```

Comment:
nit: this could also be either `std::vector<uint8_t>` or a `std::string` so that you don't have to manage allocation.